### ISSUE TITLE
Improve training loop

### ISSUE THREAD

**Issue by nmarzz** [nmarzz/aphynity#3]
Currently only trains and tests on train set.

Should include
- save best model on val set
- display graph on test set
- add adaptively constrained optimization 
- try to lower error tolerances
